src/test/java/io/bf2/kafka/authorizer/CustomAclAuthorizerTest.java

```diff
@@ -236,7 +243,44 @@ void testCreateAclsDeniedForInvalidPrincipal(String title, String principal) thr
                     .map(AclCreateResult::exception)
                     .map(Optional::get)
                     .allMatch(e -> e instanceof ApiException
-                            && CustomAclAuthorizer.CREATE_ACL_INVALID_PRINCIPAL.equals(e.getMessage())));
+                            && error.equals(e.getMessage())));
+        }
+    }
+
+    @Test
+    void testCreateAclsAllowedWhenPrincipalGivesThemselvesACLRules() throws IOException {
+        final List<CompletableFuture<AclCreateResult>> creationResults = new ArrayList<>();
+        final AclCreateResult value = new AclCreateResult(null);
+        creationResults.add(CompletableFuture.completedFuture(value));
+        creationResults.add(CompletableFuture.completedFuture(value));
+        Mockito.doReturn(creationResults).when(delegate).createAcls(any(),anyList());
+
+        try (CustomAclAuthorizer auth = new CustomAclAuthorizer(this.delegate)) {
+            auth.configure(config);
+
+            AuthorizableRequestContext rc = Mockito.mock(AuthorizableRequestContext.class);
+            Mockito.when(rc.clientAddress()).thenReturn(InetAddress.getLoopbackAddress());
+            Mockito.when(rc.listenerName()).thenReturn("security-9095");
+            final String userName = "tom";
+            Mockito.when(rc.principal()).thenReturn(new KafkaPrincipal(KafkaPrincipal.USER_TYPE, userName));
+            Mockito.when(rc.requestType()).thenReturn((int) ApiKeys.CREATE_ACLS.id);
```

Comment:
Feels like an opportunity to pull out a standard method to stub a requestContext

Reply:
Pulled out `createMockRequestContext`. Have also added a common `createAclsAndExpect` method for the 3 `createAcl` test cases since they all repeated the same work of sending through a request with some bindings then checking the create result.